client/components/app.jsx

```diff
@@ -1,12 +1,63 @@
 import React from 'react';
 import Header from './header';
+import ProductList from './product-list';
+import Table from './table';
 
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: {
+        name: 'homepage',
+        params: {}
+      },
+      product: []
+    };
+    this.setView = this.setView.bind(this);
+    this.getProductList = this.getProductList.bind(this);
+  }
+
+  setView(name, params) {
+    this.setState({
+      view: {
+        name: name,
+        params: params
+      }
+    });
+  }
+
+  componentDidMount() {
+    this.getProductList();
+  }
+
+  getProductList() {
+    fetch('/api/products')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({ product: data });
+      }
+      )
+      .catch(err => console.error(err));
+  }
+
   render() {
-    return (
-      <div>
-        <Header/>
-      </div>
-    );
+    if (this.state.view.name === 'homepage') {
+      return (
+        <div>
+          <Header setView={this.setView}/>
+        </div>
+      );
+    }
+
+    if (this.state.view.name === 'productList') {
+      return (
+        <div>
+          <Header/>
+          <ProductList setView={this.setView} />
+          <Table product={this.state.product}/>
```

Comment:
we might have many of tables in future features, here we can only include ProductList, then have Table inside ProductList

`<ProductList setView={this.setView} product={this.state.product} />`